exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillConnectionImpl.java

```diff
@@ -760,6 +765,24 @@ public void abort(Executor executor) throws SQLException {
     }
   }
 
+  @Override
+  public boolean useAnsiQuotedIdentifiers() throws SQLException {
+    boolean systemOption = false;
+    Boolean sessionOption = null;
+    String sql = String.format("select type, bool_val from sys.options where name = '%s'",
+        PlannerSettings.ANSI_QUOTES_KEY);
+    ResultSet rs = executeSql(sql);
```

Comment:
the statement is not closed, same for the result set, so resources are leaked.

Also, there's an ongoing effort to remove hand-made requests to query metadata and replace those by rpc calls. Shouldn't we introduce a new RPC call to query session state instead? (it also makes things easier for the C++ connector)